test/controllers/project_category_controller_test.exs

```diff
@@ -37,25 +37,22 @@ defmodule CodeCorps.ProjectCategoryControllerTest do
       project_category_2 = insert(:project_category, project: project, category: society)
       insert(:project_category, project: project, category: technology)
 
-      json =
+      response =
         conn
         |> get("project-categories/?filter[id]=#{project_category_1.id},#{project_category_2.id}")
         |> json_response(200)
-      data = json["data"]
-      assert length(data) == 2
-      [first_result, second_result | _] = data
-
-      assert first_result["id"] == "#{project_category_1.id}"
-      assert first_result["relationships"]["project"]["data"]["id"] == "#{project.id}"
-      assert first_result["relationships"]["project"]["data"]["type"] == "project"
-      assert first_result["relationships"]["category"]["data"]["id"] == "#{arts.id}"
-      assert first_result["relationships"]["category"]["data"]["type"] == "category"
-
-      assert second_result["id"] == "#{project_category_2.id}"
-      assert second_result["relationships"]["project"]["data"]["id"] == "#{project.id}"
-      assert second_result["relationships"]["project"]["data"]["type"] == "project"
-      assert second_result["relationships"]["category"]["data"]["id"] == "#{society.id}"
-      assert second_result["relationships"]["category"]["data"]["type"] == "category"
+
+      [first_result, second_result] = response |> Map.get("data")
+
+      first_result
+      |> assert_result_id(project_category_1.id)
+      |> assert_jsonapi_relationship("project", project.id)
+      |> assert_jsonapi_relationship("category", arts.id)
```

Comment:
Since this method checks for the type, we shouldn't have to explicitly assert for type.  This method essentially tests the type and id for us.